src/AddRecipeForm.js

```diff
@@ -0,0 +1,62 @@
+import { useState } from "react"
+import UseInputState from "./Hooks/useInputState"
+import Button from "./Style/Button"
+import Input from "./Style/Input"
+import Label from "./Style/Label"
+import { v4 as uuid } from 'uuid';
+
+function AddRecipeForm({ addRecipe, toggle }) {
+    const [nameVal, nameChange, nameReset] = UseInputState("")
+    const [descVal, descChange, descReset] = UseInputState("")
+    const [newIngredient, newIngredientChange, newIngredientReset] = UseInputState("")
+    const [ingredients, setIngredients] = useState([])
+
+    const handleRemove = (e) => {
+        const newIngredients = ingredients.filter(ing => ing.name !== e.target.name)
+        setIngredients(newIngredients)
+    }
+
+    const handleAdd = () => {
+        if (newIngredient) {
+            setIngredients([...ingredients, { "name": newIngredient }])
+            newIngredientReset()
+        }
+    }
+
+    return <form onSubmit={e => {
+        e.preventDefault()
+        if (nameVal.trim() && descVal.trim() && ingredients.length > 0) {
+            const newRecipe = { "id": uuid(), "name": nameVal, "description": descVal, "ingredients": ingredients }
+            addRecipe(newRecipe);
+            nameReset();
+            descReset();
+            newIngredientReset();
+            setIngredients([]);
+            toggle();
+        } else {
+            alert("Ops! Seems like you have an empty field!")
+        }
```

Comment:
This function is quite complex, it would be better to split it up into a separate function outside of the callback.